Dawntrail Update

WitchyLizard opened the pull request:
_Be warned I'm still very much learning rust, the free company section especially should be changed by somebody who knows what they're doing_

- Added Viera/Hrothgar races/clans
- Added SGE, RPR, VPR, PCT classes
- Added Materia/Shadow data centers

- Added/Moved worlds to match current changes
- Fixed world name parsing, as the data center is now included in the text field
  ex: `"Jenova [Aether]"`
  _todo: take advantage of this to include an easy data center parse_

- Fixed the FC parsing, as the css tag had changed and extra text is now included in the text field
  ex: `"Free CompanyFC_NAME"`

- Added title parsing to the profile scraper

WitchyLizard committed:
added new races/clans/worlds/dc, and title retrieval

WitchyLizard committed:
fixed typo

Roughsketch commented:
Thanks for the PR!

This crate hasn't really been updated for a very very long time (looks like last publish was June 26th, 2020). Because of that and since I have no tests for this repo, I'm going to just do a visual pass and assume you've checked it for now. If this gets published and breaks I'll have to take a look at it, but this isn't a super highly used crate anyway so I'm not too worried about accidental breaks (also the current one doesn't work anyway).

Just a note for me, but in the future it might be nice to have some tests to check responses maybe once a week or something to catch breaking changes.

Roughsketch commented on src/model/server.rs:
```diff
@@ -73,103 +41,219 @@ pub enum Server {
             "LOUISOIX" => Ok(Server::Louisoix),
             "MOOGLE" => Ok(Server::Moogle),
             "OMEGA" => Ok(Server::Omega),
+            "Phantom" => Ok(Server::Phantom),
```
Is this really supposed to be title case here? (Same for the fmt display for Server).

WitchyLizard committed:
fixed phantom/maduin + matched original case

WitchyLizard committed:
added datacenter parse, fix for GP/CP replacing MP css

WitchyLizard commented:
fixed the case issue, went over just incase and found a typo in Maduin/Anima, and made the fmt match your original case 

added the datacenter parse from `Server [Datacenter]` string 
i just made server/datacenter mimick how race/clan/gender was handled

also noticed as i was on fisher, GP/CP for dol/doh replaces the MP css tag now, failing the ensure
fixed to look for any MP/GP/CP css and return under MP. 
almost split them all up, but as it effectively is the same field, so just kept it simple

thanks for leaving it up, even if untouched. im running it from a serenity discord bot to learn rust, its held up pretty well. thankfully the lodestone hasnt changed much
the search builder worked as far as i tested, on multiple and a single result
profile id works as well for me now

---

also if you'd rather forget about it, ill refrain from bugging you and just keep a fork active on my account <3

Roughsketch commented:
> also if you'd rather forget about it, ill refrain from bugging you and just keep a fork active on my account <3

You're good, I can still do minor reviews and publish the updated package. If you're done with this PR I can maybe review it tonight (if I remember, I chronically forget stuff) and update the crate.

WitchyLizard commented:
> If you're done with this PR I can maybe review it tonight (if I remember, I chronically forget stuff) and update the crate.

did a once over, seems it's all set
I haven't run your util test, as i'm running it in a weird way, but it's atleast been in a running state now for my uses so far

---

- i did check out #7 , but a mini test just added latency, was easier to cache it outside. still a good idea but i wont touch it yet
- #3 was tempting, maybe a separate future pr (mainly for the Current class/Current class level/Currently worn items)
problem is with how the lodestone is setup for mounts/minions it requires another url request, and still parsing all the info
fix for now... is to give up and use [ffxivcollect's api](https://documenter.getpostman.com/view/1779678/TzXzDHM1#de1af1f4-e67c-45e1-a268-28707f545c93) for those fields, as it supplies owned/total fields for both with a cache built in

Roughsketch reviewed (approved):
LGTM, thank you very much for the PR. I'll publish this when I can, probably sometime tomorrow.
